File: src/main/java/org/gga/graph/search/dfs/TimeStamper.java
package org.gga.graph.search.dfs;

import org.gga.graph.Graph;

/**
 * @author mike
 */
public class TimeStamper extends VertexEventVisitor {
    private int time = 0;
    private final int[] timeStamps;
    private final VertexEvent event;

    public TimeStamper(int[] timeStamps, VertexEvent event) {
        this.timeStamps = timeStamps;
        this.event = event;
    }

    @Override
    public void onEvent(int v, Graph graph, VertexEvent evt) {
        if (evt == event) {
            timeStamps[v] = time;
            time++;
        }
    }
}

File: src/main/scala/org/gga/graph/search/dfs/TimeStamper.scala
package org.gga.graph.search.dfs

import org.gga.graph.search.dfs.VertexEventVisitor.VertexEvent
import org.gga.graph.Graph

/**
 * @author mike
 */
class TimeStamper(aTimeStamps: Array[Int], anEvent: VertexEvent.Value) extends VertexEventVisitor {
  private var time: Int = 0
  private final val timeStamps: Array[Int] = aTimeStamps
  private final val event: VertexEvent.Value = anEvent

  def onEvent(v: Int, graph: Graph, evt: VertexEvent.Value) {
    if (evt == event) {
      timeStamps(v) = time
      time += 1
    }
  }

}